src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/AwsDatasetModification.kt

```diff
@@ -132,6 +132,13 @@ class AwsDatasetModification private constructor(
                 )
                 .build()
         }
+        private var customDatasetSourceFactory: CustomDatasetSourceFactory = object : CustomDatasetSourceFactory {
+            override fun createCustomDatasetSource(
+                jiraHome: RemoteLocation,
+                database: RemoteLocation
+            ) = CustomDatasetSource.Builder(jiraHome, database)
```

Comment:
I was thinking about a `Consumer<CustomDatasetSource.Builder>` instead of a factory. It would make wrapping `jiraHome` and `database` harder, but it's already possible via `JiraFormula` via `DatasetHost`.

Reply:
Sounds like a multi-layer builder pattern that I was exploring in our internal analytics module. I will be honest that I'm not the biggest fan of using builders in contexts other than as a replacement for ton of different constructors, however this pattern may be unavoidable in a certain scale of software. 

Anyway I trust your judgement as an JPT architect.

Reply:
Factories are also _mainly_ to replace constructors. But as we can see here, both factories and builders are useful in other circumstances too.

Reply:
`datasetSourceConfig` sounds good